pkg/apis/openebs/v1alpha1/blockdevice_types.go

```diff
@@ -29,6 +29,8 @@ import (
 
 // DeviceSpec defines the desired state of BlockDevice
 type DeviceSpec struct {
+	// NodeAttributes has the details of the node in which this BD is attached
+	NodeAttributes  NodeAttribute       `json:"nodeAttributes"`
```

Comment:
Where do we say - omit if empty?

Reply:
`nodeAttributes.nodeName` should always be there in the spec. This why no `omitempty` tag is given